cli/analyzePath.js

```diff
@@ -0,0 +1,51 @@
+import { stat } from "fs/promises"
+import { relative, resolve, sep } from "path"
+import fileExists from "../lib/fileExists.js"
+
+export default async function analyzePath(path, { noRoot = false } = {}) {
+  if (!path) {
+    throw new Error(`Path argument is obligatory.`)
+  }
+
+  const absolutePath = resolve(path)
+
+  const root = await searchRootFolder(path)
+
+  if (!root) {
+    if (!noRoot) {
+      throw new Error(`Cannot analyze '${path}' folder`)
+    }
+  } else {
+    if (noRoot) {
+      throw new Error(`This path already has a Tiden project.`)
+    }
+  }
```

Comment:
```
const unknownPath = !root && !noRoot
const existedProject = root && noRoot
if (unknownPath) throw error;
if (existedProject) throw error;
```
just my thought will it be more readable?

Reply:
This root/noRoot stuff is making my eyes bleed.. both the way I wrote it and your suggestion. Wondering if there's maybe a third way to improve it?

It's not that the path is unknown, but rather that while using `init` command we actually want to make sure that there isn't a Tiden project there already (it would overwrite it).

But on the other hand, all other commands require a Tiden project to be there. Otherwise, user probably made a mistake (he ran the command in a bad place unintentionally).

It takes me two paragraphs to explain this. lol :)

Reply:
I added some comments to my code now. Do you think it is enough, maybe? Ideally, people should be able to understand the code without having me explain it.